File: credit-engine-api/src/main/java/com/credit/engine/application/dto/pricing/PricingSimulationResponse.java
package com.credit.engine.application.dto.pricing;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

public record PricingSimulationResponse(
        UUID receivableId,
        LocalDate valuationDate,
        BigDecimal baseRate,
        BigDecimal spreadRate,
        BigDecimal term,
        BigDecimal faceValue,
        BigDecimal discountAmount,
        BigDecimal presentValue,

        String currencyCode,
        String targetCurrencyCode,
        BigDecimal exchangeRateUsed,
        BigDecimal convertedAmount
) {
}

File: credit-engine-ui/src/services/pricing/dto/PricingSimulationResponse.ts
export default interface PricingSimulationResponse {
    receivableId: string;
    valuationDate: Date;
    baseRate: number;
    spreadRate: number;
    termMonths: number;
    faceValue: number;
    discountAmount: number;
    presentValue: number;
    currencyCode: string;
    targetCurrencyCode: string;
    exchangeRateUsed: number;
    convertedAmount: number;
}
